docs/source/tutorial/basic.rst

```diff
@@ -451,65 +504,50 @@ So instead of using the above example, we will use this predefined Classifier ch
    >>> optimizer = optimizers.SGD()
    >>> optimizer.setup(model)
 
-Finally, we can write a learning loop as following:
-
-.. testcode::
-   :hide:
-
-   datasize = 600
+Now we can build a trainer object.
 
 .. doctest::
 
-   >>> batchsize = 100
-   >>> datasize = 60000  #doctest: +SKIP
-   >>> for epoch in range(20):
-   ...     print('epoch %d' % epoch)
-   ...     indexes = np.random.permutation(datasize)
-   ...     for i in range(0, datasize, batchsize):
-   ...         x = Variable(x_train[indexes[i : i + batchsize]])
-   ...         t = Variable(y_train[indexes[i : i + batchsize]])
-   ...         optimizer.update(model, x, t)
-   epoch 0...
+   >>> updater = training.StandardUpdater(train_iter, optimizer)
+   >>> trainer = training.Trainer(updater, (20, 'epoch'), out='result')
 
-Only the last three lines are the code related to Chainer, which are already described above.
-Note that, in the last line, we pass ``model`` as a loss function.
+The second argument ``20, 'epoch'`` represents the duration of training.
```

Comment:
Actually parentheses are "syntax artifact", I mean only ``20, 'epoch'`` means it is a tuple, but ok, your version seems more readable.